### PR TITLE
doc(app-security): how to use the admin ui (AEROGEAR-8900)

### PR DESCRIPTION
**Pull request by damienomurchu**
## Motivation
https://issues.jboss.org/browse/AEROGEAR-8900

## What
Document how to use the admin ui console for the app-security project

## Why
Requirement for project

## How
Add to the app-security section

### Setup & serve site

- clone the project 
- checkout my branch AEROGEAR-8900
- change directory into the root of the project 
- need node 8 
- install antora
```bash
npm i -g @antora/cli@1.0
npm i -g @antora/site-generator-default@1.0
mkdir build
```
- build & serve site locally:
```
npm install
npm run build:local
npm run serve
```

### Verification

- Verify added content. Appears under `Monitoring App Security`:

Screenshot from 2019-04-02 12-36-23

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member 

## Progress

- [x] Finished task

### CODE REVIEW

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 1]
```diff
@@ -0,0 +1,58 @@
+= Accessing the console
+
+In order to view the {app-security-service} Admin console open the URL in a browser. The URL can be retrieved from the OpenShift console.
```
Comment (damienomurchu) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: `The URL can be retrieved from the OpenShift console.` needs to go, suggestions welcome

Comment (finp) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: after binding, you can nav from the mobile client (or whatever it's called these days)?

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 2]
```diff
@@ -0,0 +1,58 @@
+= Accessing the dashboard
+
+The {app-security-service} Admin console is viewed by opening the console URL in a browser. This URL is seen in the list of available dashboards after the {app-security-service} Service has been provisioned.
```
Comment (damienomurchu) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: @davidffrench could you let me know if this text is suitable

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc (hunk 1 above)
Comment (damienomurchu) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: I've modified this @finp & switched from using `console` to `dashboard` to be more consistent with the other mobile services.

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc (hunk 2 above)
Comment (davidffrench) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: This looks good to me @damienomurchu . Unless it is called out specifically in other parts of the documentation, I would extend this slightly to say `available services in MDC after`

Comment (damienomurchu) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: :+1:

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc (hunk 1 above)
Comment (finp) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: @damienomurchu  not sure if we're very consistent, ping me to discuss

**Comment by damienomurchu** [comment]
@finp thanks for clarification - pr updated & usage of `dashboard` replaced with `console`

**Comment by craicoverflow** [comment]
Two further steps should be included in the verification steps:

1. `npm install`
2. `npm run serve`

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+* *App Version*: The name of the application.
+* *Current Installs*: Total number of currently installed versions.
+* *Launches*: Total number of launches of this version of the application.
+* *Last launched*: The last time this application version was launched.
+* *Disable on Startup*: Whether this application version is disabled on startup.
+* *Custom Disable Message*: A custom message that can be included to note the reason this application version has been disabled.
+
+A breadcrumb link at the top of the view links back to the previous screen.
+
+== _Enabling/ Disabling an application version_
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]:
```suggestion
== Enabling/ Disabling an application version
```

I don't see italics used in other parts of the docs (see **Setting Up the App Security Service**).

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+* toggle the checkbox for that version in the *Disable on Startup* column. 
+* confirm these changes by clicking the *Save* button to persist these changes. 
+
+Navigating away from this screen with unsaved changes prompts you to save or discard these changes.
+
+To disable all versions of an application:
+
+* click the *Disable App* button
+
+== _Adding/ updating a custom disabled message_
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]:
```suggestion
== Adding/ updating a custom disabled message
```

I don't see italics used in other parts of the docs (see **Setting Up the App Security Service**).

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+To add or update a custom message for the version(s) of an application:
+
+* enter the message in the text field for that version under the *Custom Disable Message* column.
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: This should be converted from an unordered list to a numbered list. It is a sequence of steps that must be followed in the correct order for the action to be successful.

File: modules/ROOT/nav.adoc [hunk 6]
```diff
@@ -74,6 +74,7 @@ include::{partialsdir}/attributes.adoc[]
 *** xref:app-security.adoc[{app-security-service}]
 **** xref:app-security.adoc#introduction[Introduction]
 **** xref:app-security.adoc#setup[Setting Up]
+**** xref:app-security.adoc#monitoring[Monitoring App Security]
```
Comment (craicoverflow) [inline modules/ROOT/nav.adoc]: This title could just be called `Monitoring`. If you look at the other links in this group they don't mention App Security, as that much is implied from the group title (App Security).

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 7]
```diff
@@ -0,0 +1,58 @@
+= Accessing the console
+
+The {app-security-service} Console is viewed by opening the console URL in a browser. This URL is seen in the list of available services in the Mobile Developer Console (MDC) after the {app-security-service} Service has been provisioned.
+
+= Console overview
+
+The {app-security-service} Console allows you to monitor your applications and the versions of those applications.
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]:
```suggestion
The {app-security-service} Console allows you to monitor applications and their respective versions.
```

Saying "allows you to monitor **your** applications" might lead a user to believe that this is for monitoring their own installed apps.

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 8]
```diff
@@ -0,0 +1,58 @@
+= Accessing the console
+
+The {app-security-service} Console is viewed by opening the console URL in a browser. This URL is seen in the list of available services in the Mobile Developer Console (MDC) after the {app-security-service} Service has been provisioned.
+
+= Console overview
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: I think this overview should mention the most important thing about this application - the fact that the user can update the disabled status and disabled message for each app.

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+
+The {app-security-service} Console allows you to monitor your applications and the versions of those applications.
+
+Opening the console presents the applications view. Clicking on an application opens a details view for that application.
+
+Below is a detailed description of each of these views and the data you can expect each to contain.
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: Below is a detailed description of each of these views and the information you can expect each to contain.

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+To enable/ disable one or more versions of the application:
+
+* toggle the checkbox for that version in the *Disable on Startup* column. 
+* confirm these changes by clicking the *Save* button to persist these changes. 
+
+Navigating away from this screen with unsaved changes prompts you to save or discard these changes.
+
+To disable all versions of an application:
+
+* click the *Disable App* button
```
Comment (craicoverflow) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: Missing full stop.

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+* *App ID*: The fully qualified and unique name of the application
+* *Deployed Versions*: Total number of currently deployed versions.
+* *Current Installs*: Total number of current installed versions of the application.
+* *Launches*: Total number of application launches.
+
+Clicking on an application opens an application details view with more detailed information.
+
+= Application details view
+
+This view lists detailed information for the versions of that application:
```
[the remark this answers is not in the document]
Comment (damienomurchu) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: @finp what is your input here? (I'm happy to be guided by convention)

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc (hunk 3 above)
Comment (damienomurchu) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: Yes, their purpose is to distinguish the sub-sections more clearly. What are your thoughts @finp - should we drop the use of italics here?

File: modules/ROOT/nav.adoc (hunk 6 above)
Comment (damienomurchu) [reply modules/ROOT/nav.adoc]: Agree, but made it `Monitoring App Security` to be consistent with the sections for other mobile services which all use `Monitoring Data Sync Service`, `Monitoring Identity Management`, et al

**Comment by damienomurchu** [comment]
Thanks for your feedback @craicoverflow, I've actioned all of your points, except changing `application` to `app` - if @finp can provide clarity on whether `application` or `app` is preferred I will steer by that and make any necessary changes then.

File: modules/ROOT/nav.adoc (hunk 6 above)
Comment (finp) [reply modules/ROOT/nav.adoc]: 1. nav.adoc is only a convenience, not reflective of our actually headings
2. should be short and snappy, so I'm inclined to Enda's comment, but 
3. it's consistent with the other services
let's leave as is for now

**Comment by finp** [comment]
image
will follow up in code comments

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 12]
```diff
@@ -0,0 +1,59 @@
+= Accessing the console
+
+The {app-security-service} Console is viewed by opening the console URL in a browser. This URL is seen in the list of available services in the Mobile Developer Console (MDC) after the {app-security-service} Service has been provisioned.
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: very passive, what are the steps i need to perform as user

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+= Console overview
+
+The {app-security-service} Console allows you to monitor applications, their respective versions, and disable versions of those applications.
+
+Opening the console presents the applications view. Clicking on an application opens a details view for that application.
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: The console lists all the applications <condition?.
Click on the name of an application to view the details for that application.

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+Below is a detailed description of each of these views and the information you can expect each to contain.
+
+= Applications view
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: not sure if this whole sections adds any info that isn't obvious to user

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+
+Clicking on an application in the table opens a view with further information and the ability to disable versions of the application.
+
+= Application details view
+
+This view provides statistics and information about each deployed version of an application:
+
+
+* *App Version*: The name of the application.
+* *Current Installs*: Total number of currently installed versions.
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: dunno what that means

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+* *Launches*: Total number of application launches.
+
+Clicking on an application in the table opens a view with further information and the ability to disable versions of the application.
+
+= Application details view
+
+This view provides statistics and information about each deployed version of an application:
+
+
+* *App Version*: The name of the application.
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: what?

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+* *Current Installs*: Total number of currently installed versions.
+* *Launches*: Total number of launches of this version of the application.
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: this version, maybe the first field is mislabelled?

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 18]
```diff
@@ -0,0 +1,59 @@
...
+* *Last launched*: The last time this application version was launched.
+* *Disable on Startup*: Whether this application version is disabled on startup.
+* *Custom Disable Message*: A custom message that can be included to note the reason this application version has been disabled.
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: might not be used for this purpose, eg could say to upgrade.
*Custom Disable Message*: A custom message that is displayed when this version of the mobile app is disabled.

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+
+A breadcrumb at the top of the view navigates back to the previous screen.
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: delete

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+
+== _Enabling/Disabling an application version_
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: no slashes and no formatting in headings

either 'Enabling and Disabling ...'
or
Enabling ...

and add note to say 'use same procedure for disabling'

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+. toggle the checkbox for that version in the *Disable on Startup* column. 
+. confirm these changes by clicking the *Save* button to persist these changes. 
+
+Navigating away from this screen with unsaved changes prompts you to save or discard these changes.
+
+To disable all versions of an application:
+
+* click the *Disable App* button.
+
+== _Adding/updating a custom disabled message_
```
Comment (finp) [inline modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: slashes and formatting

**Comment by damienomurchu** [comment]
Thanks for all feedback @finp & @austincunningham, I will be look to action all items tomorrow (just closing out another task first)

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc (hunk 16 above)
Comment (damienomurchu) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: corrected now

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc (hunk 15 above)
Comment (damienomurchu) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: expanded upon now

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc (hunk 20 above)
Comment (damienomurchu) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: Updated the title @finp as suggested. Not sure if your point also refers to the paragraph text after the title, but have left that the same. If there's any changes needed there can you suggest

File: modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc (hunk 14 above)
Comment (damienomurchu) [reply modules/ROOT/pages/_partials/app-security/using-the-admin-ui.adoc]: @finp I've removed this whole section now :+1:

**Comment by damienomurchu** [comment]
@finp @austincunningham feedback actioned now, if you could re-review & confirm the changes are ok

@finp I think I captured all your feedback in the comments left, but not sure I captured everything in the screenshot you left, so if I've missed anything could you let me know. Ping me offline if bluejeans is easier to clarify any requested changes :)

**Comment by damienomurchu** [comment]
Thanks for all the feedback everyone, going to merge this now

**Pull request merged**